packages/web-ui/src/sections/DocumentEditor/index.tsx

```diff
@@ -0,0 +1,34 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+
+import { ConversationMetadata, readMetadata } from '@latitude-data/compiler'
+import { DocumentTextEditor, Text } from '$ui/ds/atoms'
+
+export function DocumentEditor({ document }: { document: string }) {
+  const [value, setValue] = useState(document)
+  const [metadata, setMetadata] = useState<ConversationMetadata>()
+  useEffect(() => {
+    readMetadata({ prompt: value }).then(setMetadata)
+  }, [value])
+
+  return (
+    <div className='flex flex-row w-full h-full gap-8'>
+      <div className='flex flex-col flex-1 gap-2'>
+        <div className='flex flex-row space-between'>
+          <Text.Mono color='foreground'>Prompt editor</Text.Mono>
+        </div>
+        <DocumentTextEditor
+          value={value}
+          metadata={metadata}
+          onChange={setValue}
+        />
+      </div>
+      <div className='flex flex-col flex-1'>
+        <div className='flex flex-row space-between'>
```

Comment:
Is amazing how having abstracted the ui elements all it remains for tailwind is to do do the flex thing